src/main/java/hudson/plugins/copyartifact/CopyArtifact.java

```diff
@@ -523,115 +524,132 @@ private static ItemGroup getItemGroup(Run<?, ?> build) {
     }
 
 
-    private boolean perform(Run src, Run<?,?> dst, String expandedFilter, @CheckForNull String expandedExcludes, FilePath targetDir, PrintStream console) throws IOException, InterruptedException {
+    private boolean perform(Run src, Run<?,?> dst, String expandedFilter, @CheckForNull String expandedExcludes, FilePath targetDir, TaskListener listener) throws IOException, InterruptedException {
+        PrintStream console = listener.getLogger();
         VirtualFile srcDir = selector.getArtifacts(src, console);
         if (srcDir == null) {
             return isOptional();  // Fail build unless copy is optional
         }
-
-        Map<String, String> fingerprints;
-        MessageDigest md5;
-        if (isFingerprintArtifacts()) {
-            fingerprints = new HashMap<>();
-            try {
-                md5 = MessageDigest.getInstance("MD5");
```

Comment:
Note that after the merge of #99, the diff of _this_ PR against `master` actually got longer, since it more closely followed the code in 1.39. (IIRC)